app/app.go

```diff
@@ -677,3 +678,25 @@ func (a *App) SetServer(srv *Server) {
 func (a *App) GetT() goi18n.TranslateFunc {
 	return a.t
 }
+
+func (a *App) DBHealthCheckWrite() error {
+	currentTime := strconv.FormatInt(time.Now().Unix(), 10)
+
+	return a.Srv().Store.System().SaveOrUpdate(&model.System{
+		Name:  a.dbHealthCheckKey(),
+		Value: currentTime,
+	})
+}
+
+func (a *App) DBHealthCheckDelete() error {
+	_, err := a.Srv().Store.System().PermanentDeleteByName(a.dbHealthCheckKey())
+	return err
+}
+
+func (a *App) dbHealthCheckKey() string {
+	return fmt.Sprintf("health_check_%s", a.GetClusterId())
+}
+
+func (a *App) CheckIntegrity() <-chan model.IntegrityCheckResult {
+	return a.Srv().Store.CheckIntegrity
```

Comment:
Here you are returning the `CheckIntegrity` function instead of the result of executing the `CheckIntegrity` function.